packages/core/echo/echo-pipeline/src/pipeline/pipeline.ts

```diff
@@ -277,6 +277,18 @@ export class Pipeline implements PipelineAccessor {
 
     this._timeframeClock.setTimeframe(timeframe);
 
+    // Cancel downloads of mutations before the cursor.
+    timeframe.frames().forEach(([key, seq]) => {
+      const feed = this._feeds.get(key);
+      if (!feed) {
+        throw new Error('Feed not found');
+      }
+      feed.setDownloading(false);
+      (feed.download({ start: seq + 1, end: -1, linear: true }) as Promise<number>).catch((err) => {
```

Comment:
1. annotate the type in feed-wrapper so you dont have to cast here.

2. add a comment about when this promise will resolve.

3. are you specifying `sparse: true` when feed is constructed?

4. consider using `undownload` instead of `setDownloading` or explain why this and not that.

5. To test this: `space.test.ts` seems like a good level. Create a space; write mutations; create an epoch; write more mutations. Now add another peer with separate feed store. Check that after replication the second peer only has the mutations after the epoch.

Reply:
6. Does feed.replicate cause the download range to change?

Reply:
1. Done
2. Done
3. Done
4. Done
5. Need to synchronize replication of feeds and epoch processing
6. in porgress